weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/callsTableColumns.tsx

```diff
@@ -0,0 +1,505 @@
+    // Should have custom timestamp filter here.
+    filterOperators: allOperators.filter(o => o.value.startsWith('(date)')),
+    sortable: true,
+    width: 100,
+    minWidth: 100,
+    maxWidth: 100,
+    renderCell: cellParams => {
+      return (
+        <Timestamp
+          value={convertISOToDate(cellParams.row.started_at).getTime() / 1000}
+          format="relative"
+        />
+      );
+    },
+  };
+  cols.push(startedAtCol);
+
+  cols.push({
+    field: 'derived.latency',
+    headerName: 'Latency',
+    width: 100,
+    minWidth: 100,
+    maxWidth: 100,
+    // Should probably have a custom filter here.
+    filterable: false,
+    sortable: false,
+    renderCell: cellParams => {
+      if (traceCallStatusCode(cellParams.row) === 'UNSET') {
+        // Call is still in progress, latency will be 0.
+        // Displaying nothing seems preferable to being misleading.
+        return null;
+      }
+      return monthRoundedTime(traceCallLatencyS(cellParams.row));
+    },
+  });
+
+  cols.forEach(col => {
+    if (col.field in userDefinedColumnWidths) {
+      col.width = userDefinedColumnWidths[col.field];
+      col.flex = 0;
+    }
+  });
+
+  return {cols, colGroupingModel: groupingModel};
+}
+/**
+ * This function maintains an ever-growing list of dynamic column names. It is used to
+ * determine which dynamic columns (e.g. attributes, inputs, outputs) are present in the
+ * table data. If we page/filter/sort we don't want to lose the columns that were present
+ * in the previous data.
+ */
+const useAllDynamicColumnNames = (
+  tableData: TraceCallSchema[],
+  shouldIgnoreColumn: (col: string) => boolean,
+  resetDep: any
+) => {
+  // 1. Maintain an ever-growing set of unique columns. It must be reset
+  // when `effectiveFilter` changes.
+  const currentDynamicColumnNames = useMemo(() => {
+    const dynamicColumns = new Set<string>();
+    tableData.forEach(row => {
+      Object.keys(row).forEach(key => {
+        if (
+          key.startsWith('attributes') ||
+          key.startsWith('inputs') ||
+          key.startsWith('output') ||
+          key.startsWith('summary')
+        ) {
+          dynamicColumns.add(key);
+        }
+      });
+    });
+    return _.sortBy([...dynamicColumns]);
+  }, [tableData]);
+
+  // Wow this is a pretty crazy idea to maintain a list of all dynamic columns
+  // so we don't blow away old ones
+  const [allDynamicColumnNames, setAllDynamicColumnNames] = useState(
+    currentDynamicColumnNames
+  );
+
+  useEffect(() => {
+    setAllDynamicColumnNames(last => {
+      const lastDynamicColumnNames = last.filter(c => {
+        if (shouldIgnoreColumn(c)) {
+          return false;
+        }
+        return true;
+      });
+
+      return _.sortBy(
+        Array.from(
+          new Set([...lastDynamicColumnNames, ...currentDynamicColumnNames])
+        )
+      );
+    });
+  }, [currentDynamicColumnNames, shouldIgnoreColumn]);
+
+  useEffect(() => {
+    if (resetDep) {
```

Comment:
Does the same thing in either case?

Reply:
yeah - i just want to trigger on changes to `resetDep`. just putting `resetDep` in the dep array is susceptible to linter removing. I commented. Perhaps there is a better way?